src/fabric.c

```diff
@@ -394,7 +394,10 @@ void fi_ini(void)
 	fi_register_provider(VERBS_INIT, NULL);
         /* Initialize the sockets provider last.  This will result in
            it being the least preferred provider. */
-	fi_register_provider(SOCKETS_INIT, NULL);
+        fi_register_provider(SOCKETS_INIT, NULL);
+#ifdef HAVE_MXM
+        fi_register_provider(MXM_INIT, NULL);
+#endif
```

Comment:
This still needs to be fixed.  The sockets provider needs to be last, and the ifdef should not be needed.